app/main.py

```diff
@@ -1,25 +1,65 @@
 from abc import ABC
+from typing import Type
 
 
 class IntegerRange:
-    pass
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: type, name: str) -> None:
+        self.public_name = name
+        self.private_name = "_" + name
+
+    def __get__(self, instance: object, owner: type) -> float:
+        return getattr(instance, self.private_name)
+
+    def __set__(self, instance: object, value: float) -> None:
+        if not isinstance(value, int):
+            raise TypeError("Value must be of type int")
+        if not (self.min_amount <= value <= self.max_amount):
+            raise ValueError("Invalid value")
```

Comment:
You should describe here why is it invalid. It will be really hard to fix an error with such message